app/components/primer/menu_component.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Primer
+  # The menu is a vertical list of navigational links.
+  class MenuComponent < Primer::Component
+    include ViewComponent::Slotable
+
+    with_slot :heading, class_name: "Heading"
+    with_slot :item, collection: true, class_name: "Item"
+
+    # @example 135|Default
+    #   <%= render(Primer::MenuComponent.new) do |component| %>
+    #     <% component.slot(:item, href: "#url", label: "Account", icon: "tools") %>
+    #     <% component.slot(:item, href: "#url", label: "Profile", icon: "person", selected: true) %>
+    #     <% component.slot(:item, href: "#url", label: "Emails", icon: "mail") %>
+    #   <% end %>
+    #
+    # @example 175|With heading
+    #   <%= render(Primer::MenuComponent.new) do |component| %>
+    #     <% component.slot(:heading) { "Heading" } %>
+    #     <% component.slot(:item, href: "#url", label: "Account") %>
+    #     <% component.slot(:item, href: "#url", label: "Profile", selected: true) %>
+    #     <% component.slot(:item, href: "#url", label: "Emails") %>
+    #   <% end %>
+    #
+    # @param kwargs [Hash] <%= link_to_style_arguments_docs %>
+    def initialize(**kwargs)
+      @kwargs = kwargs
+      @kwargs[:tag] = :nav
+      @kwargs[:classes] = class_names(kwargs[:classes], "menu")
+    end
+
+    def before_render
+      @kwargs['aria-labelledby'] = heading.kwargs[:id] if heading.present?
+    end
+
+    class Heading < Primer::Slot
```

Comment:
I was curious what it would look like with v2 + the lambda proposal, and I think we could get away with the following:

```ruby
renders_one :heading, -> (id: "menu-heading", tag: :span, classes: "") do
  Primer::BaseComponent.new(id: id, tag: tag, classes: class_names(classes, "menu)
end
```

Then the view could be changed from:

```erb
  <% if heading %>
    <%= render Primer::BaseComponent.new(**heading.kwargs) do %>
      <%= heading.content %>
    <% end %>
  <% end %>
```

to

```erb
<%= heading if heading.present? %>
```

Since it looks like the slot is essentially delegating to a component, we could remove some boilerplate with the above approach.